pkg/client/unit.go

```diff
@@ -264,7 +265,7 @@ func (u *Unit) updateState(exp UnitState, logLevel UnitLogLevel, cfg *proto.Unit
 	}
 	if u.configIdx != cfgIdx {
 		u.configIdx = cfgIdx
-		if u.config != cfg {
+		if !gproto.Equal(u.config.Source.ProtoReflect().Interface(), cfg.Source.ProtoReflect().Interface()) {
```

Comment:
```suggestion
		if !gproto.Equal(u.config.Source, cfg.Source) {
```

Shouldn't this work? Why is the additional `.ProtoReflect().Interface()` call needed? I just did a quick test and this builds, but I didn't run it though the tests.

Reply:
Yep tests still pass with this change.

Reply:
Ah, my bad, was poking around the `proto` library a bit, forgot to revert that.